### PR TITLE
Modify match type evaluation only

### PR DESCRIPTION
**Pull request by KnightNiwrem**

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
What was the purpose of this again?

File: src/composer.ts [hunk 1]
```diff
@@ -827,11 +827,17 @@ function triggerFn(trigger: MaybeArray<string | RegExp>) {
...
+            ? Extract<C[K], string | RegExpMatchArray>
+            : C[K];
+    },
     ":text" | ":caption"
 >;
 type CommandContext<C extends Context> = Filter<
-    C & { match: string },
+    {
+        [K in keyof C]: K extends "match" ? Extract<C[K], string> : C[K];
+    },
```
Comment (KnorpelSenf) [inline src/composer.ts:840]: Does this work?
```suggestion
    Omit<C, 'match'> & { match: Extract<C['match'], string> },
```

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]
I have one idea how this could be simplified, can you confirm that it works?

File: src/composer.ts (hunk 1 above)
Comment (KnightNiwrem) [reply src/composer.ts:840]: Verified and accepted

Comment (KnorpelSenf) [reply src/composer.ts]: Tremendous

File: src/context.ts [hunk 2]
```diff
@@ -74,7 +74,7 @@ export class Context implements RenamedUpdate {
      * Used by some middleware to store information about how a certain string
      * or regular expression was matched.
      */
-    public match?: string | RegExpMatchArray;
+    public match: string | RegExpMatchArray | undefined;
```
Comment (KnorpelSenf) [inline src/context.ts:77]: If I'm not mistaken, this can be reverted now.

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]

File: src/context.ts (hunk 2 above)
Comment (KnorpelSenf) [reply src/context.ts:77]: Because we no longer rely on a mapped type which required the field to always be present (even though it isn't).

Comment (KnorpelSenf) [reply src/context.ts:77]:
```suggestion
    public match?: string | RegExpMatchArray;
```

Comment (KnightNiwrem) [reply src/context.ts]: This is correct

**Review by KnorpelSenf — APPROVED** [review]
Thank you so much!

**Pull request merged**